src/patchTypescript.ts

```diff
@@ -0,0 +1,39 @@
+// tslint:disable-next-line:no-implicit-dependencies
+import * as ts from 'typescript'; // Imported for types alone
+
+export interface TypeScriptPatchConfig {
+  /**
+   * Ususally, the compilerHost created with typescript.createWatchCompilerHost will bail out of diagnostics collection if there has been any syntactic error.
+   * (see https://github.com/Microsoft/TypeScript/blob/89386ddda7dafc63cb35560e05412487f47cc267/src/compiler/watch.ts#L141 )
+   * If this plugin is running with `checkSyntacticErrors: false`, this might lead to situations where no syntactic errors are reported within webpack
+   * (because the file causing a syntactic error might not get processed by ts-loader), but there are semantic errors that would be missed due to this behavior.
+   * This ensures that the compilerHost always assumes that there were no syntactic errors to be found and continues to check for semantic errors.
+   */
+  skipGetSyntacticDiagnostics: boolean;
+}
+
+export function patchTypescript(
+  typescript: typeof ts,
+  config: TypeScriptPatchConfig
+) {
+  const origTypescript = { ...typescript };
```

Comment:
Should we copy entire typescript object? I think it would be more elegant to copy just methods that we will use in override :)
```
const { 
  createEmitAndSemanticDiagnosticsBuilderProgram: originalCreateEmitAndSemanticDiagnosticsBuilderProgram
} = typescript;
```

Reply:
For now yes, but #231 uses this pattern for about 20 methods (the file there is currently named wrapTypescript, I'll rename that there to be the same file as this one), so we already know that going forward, this will make a lot of method copies and take up like 20-40 lines (depending on max char width) that way.
The spread on the other hand is just an object full of references to the original implementations - what isn't overridden doesn't really take extra memory, so I'd suggest to go with this from the beginning.

Reply:
PS: actually, this exact method would need to be overwritten/wrapped a second time for 231, so I guess what I said here doesn't apply there. Gonna move it into a single variable ;)

Reply:
done as requested ;)